nvflare/fuel/utils/hash_utils.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import hashlib
+
+# A large prime number as virtual hash table size
+PRIME = 100003
+
+
+class UniformHash:
+    """A hash algorithm with uniform distribution. It achieves this with following steps,
+    1. Get a hash value using SHA256
+    2. Map the hash value to a virtual hash table
+    3. Map the virtual bucket to real bucket
+
+    """
+
+    def __init__(self, num_buckets: int):
+        self.num_buckets = num_buckets
```

Comment:
Add sanity check against num_buckets: should be a positive int less than a max.

Reply:
Done